lib/galaxy/util/__init__.py

```diff
@@ -235,19 +236,10 @@ def parse_xml_string(xml_string):
 
 def xml_to_string(elem, pretty=False):
     """Returns a string from an xml tree"""
+    xml_str = ElementTree.tostring(elem)
     if pretty:
-        elem = pretty_print_xml(elem)
-    try:
-        if PY2:
-            return ElementTree.tostring(elem)
-        else:
-            return ElementTree.tostring(elem, encoding='unicode')
-    except TypeError as e:
-        # we assume this is a comment
-        if hasattr(elem, 'text'):
-            return "<!-- %s -->\n" % (elem.text)
-        else:
-            raise e
+        pretty_string = xml.dom.minidom.parseString(xml_str).toprettyxml(indent='    ')
+        return "\n".join([line for line in pretty_string.split('\n') if not re.match(r'^[\s\\nb\']*$', line)])
```

Comment:
This method now does nothing if `pretty` is `False.

Reply:
Oops, thanks, I fixed that in a rebase